src/clusterfuzz/_internal/cron/external_testcase_reader.py

```diff
@@ -0,0 +1,149 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Automated ingestion of testcases via IssueTracker."""
+
+import re
+
+import requests
+
+from appengine.libs import form
+from appengine.libs import gcs
+from appengine.libs import helpers
+from clusterfuzz._internal.issue_management.google_issue_tracker import \
+    issue_tracker
+
+ACCEPTED_FILETYPES = [
+    'text/javascript', 'application/pdf', 'text/html', 'application/zip'
+]
+
+
+def close_invalid_issue(upload_request, attachment_info, description):
+  """Closes any invalid upload requests with a helpful message."""
+  comment_messsage = (
+      'Hello, this issue is automatically closed. Please file a new bug after'
+      'fixing the following issues:\n\n')
+  invalid = False
+
+  # TODO(pgrace) remove after testing.
+  if upload_request.id == '373893311':
+    return False
+
+  # TODO(pgrace) add secondary check for authorized reporters.
+
+  # Issue must have exactly one attachment.
+  if len(attachment_info) != 1:
+    comment_messsage += 'Please provide exactly one attachment.\n'
+    invalid = True
+  else:
+    # Issue must use one of the supported testcase file types.
+    if attachment_info[0]['contentType'] not in ACCEPTED_FILETYPES:
+      comment_messsage += (
+          'Please provide an attachment of type: html, js, pdf, or zip.\n')
+      invalid = True
+    if not attachment_info[0]['attachmentDataRef'] or \
+      not attachment_info[0]['attachmentDataRef']['resourceName'] \
+        or not attachment_info[0]['filename']:
+      comment_messsage += \
+        'Please check that the attachment uploaded successfully.\n'
+      invalid = True
+
+  # Issue must have valid flags as the description.
+  flag_format = re.compile(r'^([ ]?\-\-[A-Za-z\-\_]*){50}$')
+  if flag_format.match(description):
+    comment_messsage += (
+        'Please provide flags in the format: "--test_flag_one --testflagtwo",\n'
+    )
+    invalid = True
+
+  if invalid:
+    comment_messsage += (
+        '\nPlease see the new bug template for more information on how to use'
+        'Clusterfuzz direct uploads.')
+    upload_request.status = 'not_reproducible'
+    upload_request.save(new_comment=comment_messsage, notify=True)
+
+  return invalid
+
+
+def submit_testcase(issue_id, file, filename, filetype, cmds):
+  """Uploads the given testcase file to Clusterfuzz."""
+  if filetype == 'text/javascript':
+    job = 'linux_asan_d8_dbg'
+  elif filetype == 'application/pdf':
```

Comment:
I wonder if a system where the user specified the job would  be better?

Reply:
mmm thats a good idea - the intention for this is to limit the flexibility/chances of misuse for the researchers/external folks to start. so I think youre right - we may want to add that control for the uploaders down the line. Is there a concern with this way or were you simply recommending a better approach?

Reply:
I don't have a concern, just suggesting an alternative.